deep_translator/base.py

```diff
@@ -177,7 +179,11 @@ def _translate_batch(self, batch: List[str], **kwargs) -> List[str]:
         if not batch:
             raise Exception("Enter your text list that you want to translate")
         arr = []
-        for i, text in enumerate(batch):
-            translated = self.translate(text, **kwargs)
+        for _i, text in enumerate(batch):
+            if text in self._already_translated.keys:
```

Comment:
is it .keys or .keys()?
btw I think you can even remove the .keys since it checks keys by default

Reply:
@nidhaloff @NorthOC , Just a thought, wouldn't it be nice to use the python builtin `lru_cache` system here ?

Reply:
@jiaulislam Great idea, yes it would be nice

Reply:
> is it .keys or .keys()? btw I think you can even remove the .keys since it checks keys by default

Removed redundant keys method on dict when iterating.



> @nidhaloff @NorthOC , Just a thought, wouldn't it be nice to use the python builtin `lru_cache` system here ?

No idea how caching works :/

Reply:
@jiaulislam @NorthOC Alright, I think it's ok to merge this. 
Maybe @jiaulislam can take a look at caching in a separate PR, if you have time of course.

Reply:
> @jiaulislam @NorthOC Alright, I think it's ok to merge this. Maybe @jiaulislam can take a look at caching in a separate PR, if you have time of course.

yes, certainly. I will do it after PR #202